test/WebJobs.Script.Tests/Configuration/ActiveHostConfigurationSourceTests.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Azure.WebJobs.Script.Configuration;
+using Microsoft.Azure.WebJobs.Script.Extensions;
+using Microsoft.Extensions.Configuration;
+using Moq;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.Configuration
+{
+    public class ActiveHostConfigurationSourceTests
+    {
+        private Mock<IScriptHostManager> _mockScriptHostManager;
+
+        public ActiveHostConfigurationSourceTests()
+        {
+            _mockScriptHostManager = new Mock<IScriptHostManager>();
+        }
+
+        [Fact]
+        public void RetrievesCorrectValue()
+        {
+            var testData = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+            {
+                { "AzureWebJobsStorage", "testValue1" },
+                { "TestSection1:SomeKey", "value1" },
+                { "TestSection1:AnotherKey", "value2" },
+                { "ConnectionStrings__ConnStr1", "str1" },
+                { "ConnectionStrings__ConnStr2", "str2" },
+                { "SectionA:SectionB", "middleValue" },
+                { "SectionA:SectionB:SectionC", "finalValue" }
+            };
+
+            using (new TestScopedEnvironmentVariable(testData))
+            {
+                var testActiveHostConfig = new ConfigurationBuilder()
+                    .AddEnvironmentVariables()
```

Comment:
Can you use `InMemory` for this instead of env vars? In general, we prefer that as it relies less upon what someone has set up on their machine. (then you don't need the TestScopedEnvVar either).